browser/ui/tabs/brave_tab_strip_model.cc

```diff
@@ -132,3 +150,38 @@ void BraveTabStripModel::SetCustomTitleForTab(
 
   NotifyTabChanged(tab_interface, TabChangeType::kAll);
 }
+
+void BraveTabStripModel::OnTreeTabRelatedPrefChanged() {
+  if (*tree_tabs_enabled_ && *vertical_tabs_enabled_) {
+    BuildTreeTabs();
+  } else {
+    FlattenTreeTabs();
+  }
+}
+
+void BraveTabStripModel::BuildTreeTabs() {
+  CHECK(base::FeatureList::IsEnabled(tabs::features::kBraveTreeTab));
+  CHECK(!in_tree_mode_);
+
+  auto* unpinned_collection = contents_data_->unpinned_collection();
+  CHECK(unpinned_collection);
+
+  TreeTabNode::BuildTreeTabs(*unpinned_collection);
+  in_tree_mode_ = true;
+}
+
+void BraveTabStripModel::FlattenTreeTabs() {
+  CHECK(base::FeatureList::IsEnabled(tabs::features::kBraveTreeTab));
+  CHECK(in_tree_mode_);
```

Comment:
In CI, check failed here in CI.
I think this should be early return instead of CHECK.
`OnTreeTabRelatedPrefChanged()` could run this method if it's already in non tree mode.